added WAYBAR_HIDDEN env var used to hide the waybar

Skerga opened the pull request:
Added a check for `WAYBAR_HIDDEN=yes` to the base-app image.

side effect is that this made it necessary to overwrite `$HOME/.config/waybar/config.jsonc` on every start to allow to switch between hidden and visible. Also every image using the base-app image will need to be rebuild.

i tested the change with a local build Lutris image and it worked fine.

ABeltramo commented:
There has to be a better way to do this without having to duplicate the full file.. I'll look to see if Waybar supports multiple config files and how it merges conflicting configs

Skerga commented:
I changed around the way i achieved hiding the waybar. I added a small python script that searches for, and changes the 3 variables i used for hiding the waybar.
its not yet perfect since there is no check if those 3 variables are in the config. if some are missing my script wont set them currently. I might also rewrite the script in the following days so it will be more robust.

shawly commented:
I just looked at this and truth be told, the current python implementation could be done with `sed` alone in a simple shell script. With the flexibility of python it would make much more sense to install a jsonc parser to modify the data, because search and replacing strings can easily break as soon as somebody changes the defaults in the waybar config.

BUT I honestly see not much reason to preserve the comments in the waybar config after it has been copied from /cfg/waybar/ to $HOME/.config/waybar/.
So there is technically a much easier solution, as we already have jq installed in our base-app image, we can simply use sed to remove the comments from the waybar config, making it parsable by jq since it's now a regular json file.

```shell
WAYBAR_CONFIG="$HOME/.config/waybar/config.jsonc"
TMP_JSON=$(mktemp)
sed '/^\s*\/\//d' "$WAYBAR_CONFIG" > "$TMP_JSON"

if [ -n "$WAYBAR_HIDDEN" ]; then
  jq '.layer = "bottom" | .mode = "hide" | .start_hidden = true' "$TMP_JSON" > "$WAYBAR_CONFIG"
else
  jq '.layer = "bottom" | .mode = "dock" | .start_hidden = false' "$TMP_JSON" > "$WAYBAR_CONFIG"
fi
```

I created a script that wraps around jq for modifying or parsing jsonc files https://github.com/shawly/gow/blob/167a2744b4adc679f70c31db8f78e3dfd3950216/images/base-app/build/scripts/jsoncq.sh

I'm currently building a new image on my fork that uses the script to modify the waybar config. As soon as I can successfully test my image I will create a PR, if the script is accepted upstream, we could use it to implement the WAYBAR_HIDDEN env in a little more streamlined way.
